OWML.ModHelper/Logging/ModSocketOutput.cs

```diff
@@ -1,59 +1,111 @@
+        [Obsolete]
+        public override void WriteLine(params object[] objects)
         {
-            Logger?.Log(s);
-            CallWriteCallback(Manifest, s);
-            var message = $"{Manifest.Name}{NameMessageSeparator}{s}";
-            InternalWriteLine(message);
+            var line = string.Join(" ", objects.Select(o => o.ToString()).ToArray());
+            var type = TypeFromContents(line);
+            WriteLine(type, line, GetCallingMethodName(new StackTrace()));
         }
 
-        public override void WriteLine(params object[] objects)
+        public override void WriteLine(string line)
+        {
+            var type = MessageType.Message;
+            WriteLine(type, line, GetCallingMethodName(new StackTrace()));
+        }
+
+        public override void WriteLine(string line, MessageType type)
+        {
+            WriteLine(type, line, GetCallingMethodName(new StackTrace()));
+        }
+
+        private void WriteLine(MessageType type, string line, string senderType)
         {
-            WriteLine(string.Join(" ", objects.Select(o => o.ToString()).ToArray()));
+            Logger?.Log(line);
+            CallWriteCallback(Manifest, line);
+
+            var message = new SocketMessage
+            {
+                SenderName = Manifest.Name,
+                SenderType = senderType,
+                Type = type,
+                Message = line
+            };
+            var json = JsonConvert.SerializeObject(message);
+
+            WriteToSocket(json);
         }
 
-        private void CreateSocket()
+        private string GetCallingMethodName(StackTrace frame)
         {
-            var consolePortArgument = CommandLineArguments.GetArgument(Constants.ConsolePortArgument);
-            if (!int.TryParse(consolePortArgument, out var port))
+            try
+            {
+                return frame.GetFrame(1).GetMethod().DeclaringType.Name;
+            }
+            catch
             {
-                Logger?.Log("Error: Missing or incorrectly formatted console port argument");
-                return;
+                return "";
             }
+        }
 
+        private void ConnectToSocket()
+        {
             _socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
-            var ipAddress = IPAddress.Parse(LocalHost);
-            var endPoint = new IPEndPoint(ipAddress, port);
+            var ipAddress = IPAddress.Parse(Constants.LocalAddress);
+            var endPoint = new IPEndPoint(ipAddress, _port);
             _socket.Connect(endPoint);
         }
 
-        private void InternalWriteLine(string message)
+        private void WriteToSocket(string message)
         {
-            var bytes = Encoding.UTF8.GetBytes(message + Environment.NewLine);
+            var bytes = Encoding.UTF8.GetBytes(message);
             _socket?.Send(bytes);
         }
+
+        private MessageType TypeFromContents(string line)
```

Comment:
I think this is no longer in use

Reply:
it's used in [Obsolete]WriteLine(params object[])

Reply:
should it be, tho? I'd expect Writeline(params object[]) to behave the same as WriteLine(string)

Reply:
removed 👍

Reply:
(default message type there too)